ui/apps/platform/src/services/PoliciesService.ts

```diff
@@ -132,57 +90,45 @@ export function enableDisableNotificationsForPolicies(
 /*
  * Save a policy.
  */
-export function savePolicy(policy: Policy): Promise<Record<string, never>> {
+export function savePolicy(policy: Policy): Promise<Empty> {
     if (!policy.id) {
         throw new Error('Policy entity must have an id to be saved');
     }
     const transformedPolicy = transformPolicyCriteriaValuesToStrings(policy);
 
     return axios
-        .put<Record<string, never>>(`${baseUrl}/${policy.id}`, transformedPolicy)
+        .put<Empty>(`${baseUrl}/${policy.id}`, transformedPolicy)
         .then((response) => response.data);
 }
 
 /*
  * Create a new policy.
  */
-export function createPolicy(policy: Policy): Promise<{ data: Policy }> {
-    /*
-     * TODO after policiesSagas.js has been deleted:
-     * function return type: Promise<Policy>
-     * add method: then((response) => response.data)
-     */
+export function createPolicy(policy: Policy): Promise<Policy> {
     const transformedPolicy = transformPolicyCriteriaValuesToStrings(policy);
 
-    return axios.post(`${baseUrl}?enableStrictValidation=true`, transformedPolicy); // TODO prop?
+    return axios
+        .post<Policy>(`${baseUrl}?enableStrictValidation=true`, transformedPolicy)
+        .then((response) => response.data);
 }
 
 /*
  * Start a dry run for a policy. Return the jobId.
  */
-export function startDryRun(policy: Policy): Promise<{ data: { jobId: string } }> {
-    /*
-     * TODO after policiesSagas.js has been deleted:
-     * function return type: Promise<string>
-     * add method: then((response) => response?.data?.jobId)
-     */
+export function startDryRun(policy: Policy): Promise<string> {
     const transformedPolicy = transformPolicyCriteriaValuesToStrings(policy);
 
-    return axios.post(`${baseUrl}/dryrunjob`, transformedPolicy);
+    return axios
+        .post<{ jobId: string }>(`${baseUrl}/dryrunjob`, transformedPolicy)
+        .then((response) => response.data.jobId);
```

Comment:
should we have the ? operator here for safety? it's mentioned in the comment -- was it intentional to leave it out?

Reply:
Good question. When I wrote the comment, I had noticed extensive use of optional chaining in service functions, but had second thoughts because an infinite polling loop of `checkDryRun` calls seems possible with empty string or undefined. Therefore, failure might not be an option for the response of this request.